src/checkout-buttons/strategies/amazon-maxo/amazon-maxo-button-strategy.spec.ts

```diff
@@ -0,0 +1,183 @@
+                    });
+            });
+
+            it('fails to create button if not PaymentMethod is supplied', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if config is not initialized', async () => {
+                jest.spyOn(store.getState().config, 'getStoreConfig').mockReturnValue(undefined);
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if merchantId is not supplied', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(getPaymentMethodMockUndefinedMerchant());
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+
+            it('initialises the payment processor once', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+                strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if not container id is supplied', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                await expect( strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if not a valid container id is supplied', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.InvalidContainer);
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let checkoutButtonOptions: CheckoutButtonInitializeOptions;
+
+        beforeEach(() => {
+            checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.Full);
+            container = document.createElement('div');
+        });
+
+        afterEach(() => {
+            document.body.appendChild(container);
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            await strategy.deinitialize();
+
+            if (checkoutButtonOptions.containerId) {
```

Comment:
you can remove these `if` conditions by doing 

```js
// tslint:disable-next-line:no-non-null-assertion
const button = document.getElementById(checkoutButtonOptions.containerId!);
```

same with the if below

Reply:
If statement was removed.